File: Python/2530.py
class Oven:
	def __init__(self, Hour, Min, Sec, Sec2):
		self._hour = Hour
		self._min = Min
		self._sec = Sec
		self._sec2 = Sec2

	def manageHour(self):
		if self._sec2 >= 3600:
			self._hour += self._sec2 // 3600
			if self._hour > 23:
				self._hour %= 24
			self._sec2 -= 3600 * (self._sec2 // 3600)

	def manageMin(self):
		self.manageHour()
		if self._sec2 >= 60:
			self._min += self._sec2 // 60
			if self._min > 59:
				self._min %= 60
				self._hour += 1
			self._sec2 -= 60 * (self._sec2 // 60)

	def manageSec(self):
		self.manageMin()
		self._sec += self._sec2
		if self._sec > 59:
			self._sec %= 60
			self._min += 1
			if self._min > 59:
				self._min %= 60
				self._hour += 1
				if self._hour > 23:
					self._hour %= 24
		print("%d %d %d" % (self._hour, self._min, self._sec))

hour, minute, second = map(int, input().split())
second2 = int(input())
Oven(hour, minute, second, second2).manageSec()

File: C++/2530.cpp
#include <iostream>
using namespace std;

class Oven
{
  private:
 	int hour;
 	int minn;
 	int sec;
 	int sec2;
 public:
 	void Initialize(int Hour, int Min, int Sec, int Sec2)
 	{
 		hour = Hour;
 		minn = Min;
 		sec = Sec;
 		sec2 = Sec2;
 	}

 	void ManageHour()
 	{
 		if (sec2 >= 3600){
 			hour += sec2 / 3600;
 			if (hour > 23){
 				hour %= 24;
 			}
 			sec2 -= 3600 * (sec2 / 3600);
 		}
 	}

 	void ManageMin()
 	{
 		if (sec2 >= 60){
 			minn += sec2 / 60;
 			if (minn > 59){
 				minn %= 60;
 				hour += 1;
 			}
 			sec2 -= 60 * (sec2 / 60);
 		}
 	}

 	void ManageSec()
 	{
 		sec += sec2;
 		if (sec > 59){
 			sec %= 60;
 			minn += 1;
 			if (minn > 59){
 				minn %= 60;
 				hour += 1;
 				if (hour > 23){
 					hour %= 24;
 				}
 			}
 		}
 		cout<<hour<<' '<<minn<<' '<<sec;
 	}
};

int main()
{
	int hour, minute, second, second2;
	cin>>hour>>minute>>second;
	cin>>second2;
	Oven answer;
	answer.Initialize(hour, minute, second, second2);
	answer.ManageHour();
	answer.ManageMin();
	answer.ManageSec();

	return 0;

}
